src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,99 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIALIZATION_LENGTH = 10;
+    private static final String ERROR_MESSAGE_INDEX = "Your index is incorrect";
+    private static final String ERROR_MESSAGE_ELEMENT = "Can't remove element for such";
+    private static final double GROW_INDEX = 1.5;
+    protected T[] array;
+    protected int size = 0;
+
+    ArrayList() {
+        array = (T[])new Object[INITIALIZATION_LENGTH];
+    }
+
     @Override
     public void add(T value) {
-
+        makeBiggerArrayIfNeeded();
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        size++;
+        checkIndex(index);
+        makeBiggerArrayIfNeeded();
+        System.arraycopy(array,index,array,index + 1, size - index);
+        array[index] = value;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return (T)array[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndex(index);
+        array[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        size--;
+        T returnValue = array[index];
+        System.arraycopy(array, index + 1, array, index, size - index);
+        return returnValue;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        for (int i = 0; i < array.length; i++) {
+            if (array[i] == null && element != null) {
+                continue;
+            }
+            if ((array[i] == null && element == null) || array[i].equals(element)) {
+                return remove(i);
+            }
```

Comment:
I guess  you can simplify  it more 
```suggestion
         
            if (array[i] ==  element ||  element != null && element.equals(array[i])) {
                return remove(i);
            }
```

Reply:
I think I cant simplify it more, this code raising error, I need to check if array[i] is null and if it is and element is not I need to continue cycle but not to return.